### ISSUE TITLE
the result of `offset` is abnormal

### ISSUE THREAD

**Issue by n4mine** [VictoriaMetrics/VictoriaMetrics#625]
**Describe the bug**
after upgrade vmselect from 6f1d92669840c79bd637ce2b8a092fd917f90c49 to 4353ff7ef13a0472a0472b389d8bb352a6770f0c
the result of `offset` is abnormal
i think it relate to  17f175ff5a7cae80e174535b227298b7f1d40c7e ?

**To Reproduce**
`q offset 24h` from `today 00:00`  to `today 24:00`, the result is odd.
from `now()` to `24:00` is a straight line

**Expected behavior**
the result of `q offset 24h` from `00:00` to `24:00` is equal to `q` from `00:00 - 24h` to `24:00 -24h`

**Screenshots**
![111](https://user-images.githubusercontent.com/3345153/87380974-30bb5000-c5c6-11ea-8a16-acc3a74538cc.jpg)

**Version**
```
$ ./vmselect-prod -version
vmselect-20200713-124700-heads-cluster-0-g4353ff7e
```

**Comment by valyala** [comment]
@n4mine , thanks for the bug report! The bug should be fixed in the commit 0e7b2008b2ee5ba56782f192d4e987a295a9f743 .

**Comment by n4mine** [comment]
confirm, thanks @valyala

**Comment by valyala** [comment]
The bugfix has been included in [v1.38.1](https://github.com/VictoriaMetrics/VictoriaMetrics/releases) release.
